EIPS/eip-721.md

```diff
@@ -0,0 +1,312 @@
+
+> You just received deed ID `0xb9d6a94f1ab16f461b2af06aebab7e1cfe10ada985da001f274f370fbb848a40` from contract `0x0dcd2f752394c41875e259e00bb44fd505297caf`!
+
+```
+/// @title Metadata extension to ERC-721 interface
+/// @author William Entriken (https://phor.net)
+/// @dev Specification at https://github.com/ethereum/eips/issues/XXXX
+interface ERC721Metadata is ERC721 {
+
+    /// @dev ERC-165 (draft) interface signature for ERC721
+    // bytes4 internal constant INTERFACE_SIGNATURE_ERC721Metadata = // 0x2a786f11
+    //     bytes4(keccak256('name()')) ^
+    //     bytes4(keccak256('symbol()')) ^
+    //     bytes4(keccak256('deedUri(uint256)'));
+
+    /// @notice A descriptive name for a collection of deeds managed by this
+    ///  contract
+    /// @dev Wallets and exchanges MAY display this to the end user.
+    function name() public pure returns (string _deedName);
+
+    /// @notice An abbreviated name for deeds managed by this contract
+    /// @dev Wallets and exchanges MAY display this to the end user.
+    function symbol() public pure returns (string _deedSymbol);
+
+    /// @notice A distinct URI (RFC 3986) for a given token.
+    /// @dev If:
+    ///  * The URI is a URL
+    ///  * The URL is accessible
+    ///  * The URL points to a valid JSON file format (ECMA-404 2nd ed.)
+    ///  * The JSON base element is an object
+    ///  then these names of the base element SHALL have special meaning:
+    ///  * "name": A string identifying the item to which `_deedId` grants
+    ///    ownership
+    ///  * "description": A string detailing the item to which `_deedId` grants
+    ///    ownership
+    ///  * "image": A URI pointing to a file of image/* mime type representing
+    ///    the item to which `_deedId` grants ownership
+    ///  Wallets and exchanges MAY display this to the end user.
+    ///  Consider making any images at a width between 320 and 1080 pixels and
+    ///  aspect ratio between 1.91:1 and 4:5 inclusive.
+    function deedUri(uint256 _deedId) external view returns (string _uri);
+}
+```
+
+A second extension, the **ERC-721 Accountability Extension**, is OPTIONAL and allows your contract to make all deeds discoverable.
+
+```solidity
+/// @title Enumeration extension to ERC-721 interface
+/// @author William Entriken (https://phor.net)
+/// @dev Specification at https://github.com/ethereum/eips/issues/XXXX
+contract ERC721Enumerable is ERC721 {
+
+    /// @dev ERC-165 (draft) interface signature for ERC721
+    // bytes4 internal constant INTERFACE_SIGNATURE_ERC721Enumerable = // 0xa5e86824
+    //     bytes4(keccak256('deedByIndex()')) ^
+    //     bytes4(keccak256('countOfOwners()')) ^
+    //     bytes4(keccak256('ownerByIndex(uint256)'));
+
+    /// @notice Enumerate active deeds
+    /// @dev Throws if `_index` >= `countOfDeeds()`
+    /// @param _index A counter between zero and `countOfDeeds()`, inclusive
```

Comment:
Is the range inclusive also at `countOfDeeds()`?

Reply:
Should be between zero and `countOfDeeds()-1`, inclusive.

Reply:
Updated in 6754461, that new language usable?